source/slang/slang-ir-inline.cpp

```diff
@@ -475,13 +589,48 @@ struct InliningPassBase
         return clonedInst;
     }
 
+    /// Insert debug scope information in the callee blocks before cloning
```

Comment:
I thought we discussed last time that the inline pass will never modify any callee blocks before cloning?

Reply:
I am not sure I understand why we need this to be in its own function. We should be able to just emit a debugScope at start of `inlineSingleBlockFuncBody`, and emit a `debugNoScope` toward the end of `inlineSingleBlockFuncBody`, right before calling `updateDebugInlineAt`.

Also I am not sure why we need separate functions for single/multi blocks to handle `updateDebugInlineAt`. If this is implemented carefully, the updateDebugInlineAt logic shouldn't care if it is single or multi-block. The multi-block logic should just apply directly to single block case.